YSI_Game/y_weapondata/y_weapondata_entry.inc

```diff
@@ -87,32 +82,42 @@ Optional plugins:
 	88   `8b d8'   88 ,adPPPPP88 88 88       88      d8""""""""8b   88          88  
 	88    `888'    88 88,    ,88 88 88       88     d8'        `8b  88          88  
 	88     `8'     88 `"8bbdP"Y8 88 88       88    d8'          `8b 88          88  
-
 */
 
-#define WEAPONDATA_NO_AMMO 	" "
-#define WEAPONDATA_45ACP 		".45 ACP"
-#define WEAPONDATA_9MM 		"9mm"
-#define WEAPONDATA_12GUAGE	"12 Guage"
-#define WEAPONDATA_556MM		"5.56mm"
-#define WEAPONDATA_762MM		"7.62mm"
-#define WEAPONDATA_300MAGNUM	".300 Magnum"
-#define WEAPONDATA_ROCKET		"Rocket"
-#define WEAPONDATA_INVALID_RETURN -1
-
-#define WEAPON_FIRE_TYPE_MELEE 0
-#define WEAPON_FIRE_TYPE_PROJECTILE 1
-#define WEAPON_FIRE_TYPE_INSTANT_HIT 2
-#define WEAPON_FIRE_TYPE_AREA_EFFECT 3
-#define WEAPON_FIRE_TYPE_USE 4
-#define WEAPON_FIRE_TYPE_CAMERA 5
-//P:D(Weapon_IsValid(weaponid));
-//#define Weapon_IsValid(%0) (%0>=1 && %0<=18 || %0>=21 && %0<=46)
-stock bool:Weapon_IsValid(weaponid)
+#define WEAPONDATA_NO_AMMO 				" "
+#define WEAPONDATA_45ACP 				".45 ACP"
+#define WEAPONDATA_9MM 					"9mm"
+#define WEAPONDATA_12GUAGE				"12 Guage"
+#define WEAPONDATA_556MM				"5.56mm"
+#define WEAPONDATA_762MM				"7.62mm"
+#define WEAPONDATA_300MAGNUM			".300 Magnum"
+#define WEAPONDATA_ROCKET				"Rocket"
+#define WEAPONDATA_INVALID_RETURN 		-1
+
+#define WEAPON_FIRE_TYPE_MELEE 			0
+#define WEAPON_FIRE_TYPE_PROJECTILE 	1
+#define WEAPON_FIRE_TYPE_INSTANT_HIT 	2
+#define WEAPON_FIRE_TYPE_AREA_EFFECT 	3
+#define WEAPON_FIRE_TYPE_USE 			4
+#define WEAPON_FIRE_TYPE_CAMERA 		5
+
+stock YSI_Weapon_IsValid__(weaponid)
 {
-	return (weaponid >= 1 && weaponid <= 18 || weaponid >= 21 && weaponid <= 46);
+	switch(weaponid)
+	{
+		case 1 .. 18:
+			return true;
+		case 21 .. 46:
+			return true;
+	}
+	return false;
 }
 
+// Many libraries already have `Weapon_IsValid` defined.
+#if !defined Weapon_IsValid
+	#define Weapon_IsValid(%0) (%0>=1&&%0<=18||%0>=21&&%0<=46)
```

Comment:
Don't use a macro for this, they are broken.

Reply:
For some reason `stock Weapon_IsValid` gives an invalid instruction run-time error. It only works if it is a macro.